modules/contracts/apps/challenge/src/handlers/execute.rs

```diff
@@ -0,0 +1,568 @@
+use crate::error::AppError;
+use abstract_dex_adapter::msg::OfferAsset;
+use abstract_sdk::features::AbstractResponse;
+use cosmwasm_std::{Addr, DepsMut, Env, MessageInfo, Response, StdError, Timestamp, Uint128};
+
+use crate::contract::{AppResult, ChallengeApp};
+use abstract_sdk::prelude::*;
+
+use crate::msg::ChallengeExecuteMsg;
+use crate::state::{
+    ChallengeEntry, ChallengeEntryUpdate, ChallengeStatus, CheckIn, CheckInStatus, DurationChoice,
+    EndType, Friend, Penalty, UpdateFriendsOpKind, Vote, ADMIN, CHALLENGE_FRIENDS, CHALLENGE_LIST,
+    CHALLENGE_VOTES, DAILY_CHECK_INS, NEXT_ID, VOTES,
+};
+
+pub fn execute_handler(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    app: ChallengeApp,
+    msg: ChallengeExecuteMsg,
+) -> AppResult {
```

Comment:
Since it seems every api is admin controlled, we *could* do the admin assertion here, though it wouldn't change semantics.

Reply:
With the exception of the friends methods, so nevermind.